superset/dataframe.py

```diff
@@ -41,26 +63,50 @@ class SupersetDataFrame(object):
         'V': None,   # raw data (void)
     }
 
-    def __init__(self, df):
-        self.__df = df.where((pd.notnull(df)), None)
+    def __init__(self, data, cursor_description, db_engine_spec):
+        column_names = []
+        if cursor_description:
+            column_names = [col[0] for col in cursor_description]
+        self.column_names = dedup(column_names)
+
+        # check whether the result set has any nested dict columns
```

Comment:
This comment confused me... we're not looking for nested dicts here, just dicts, right?

Reply:
I just moved that code from `sql_lab.convert_results_to_df`, but I think you're right about the comment. I didn't write that line originally.